server/text_generation_server/models/custom_modeling/flash_llama_modeling.py

```diff
@@ -632,13 +630,12 @@ class FlashLlamaForCausalLM(torch.nn.Module):
     def __init__(self, prefix: str, config, weights):
         super().__init__()
 
+        if config.model_type == "mllama_text_model":
+            prefix = f"{prefix}.model"
```

Comment:
No. The correct line was whatever line was before.

This class cannot know about the model_type (shouldn't).
Especially since you're removing everything a few lines below.

No shenanigans here.

Reply:
100% agreed that the class should not know about the model type, I've updated the logic to handle this case by avoiding appending `.model` if the prefix ends in `text_model`

```python
base_model = "" if prefix.endswith("text_model") else ".model"
```

The reason for this complexity is the naming convention used by idefics3. The model has weights with names like `model.text_model.embed_tokens.weight` and the current logic always expects models to contain `model.embed_tokens` or `X.model.embed_tokens`.

The latest changes handle this case by conditionally appending `".model"` before constructing the prefixes. Please let me know if theres a better way to handle this 🙏